.infrastructure/cronjob.yml

```
@@ -0,0 +1,31 @@
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: cronjob
+  namespace: mateapp
+spec:
+  schedule: "*/4 * * * *"
+  concurrencyPolicy: Allow
+  startingDeadlineSeconds: 30
+  successfulJobHistoryLimit: 10
+  failedJobsHistoryLimit: 5
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          containers:
+            - name: health-check
+              image: radial/busyboxplus:curl
+              resources:
+                requests:
+                  cpu: "100m"
+                  memory: "128Mi"
+                limits:
+                  cpu: "250m"
+                  memory: "256Mi"
+              command: ["/bin/sh", "-c"]
+              args:
+                - >
+                  echo "Checking todoapp /api/health endpoint...";
+                  curl -s http://todoapp-service/api/health || echo "Failed to reach todoapp"
```

The current `args` line uses `curl -s http://todoapp-service/api/health || echo "Failed to reach todoapp"`. This construction masks failure exit codes because the `echo` will return 0 and the pod will succeed even if `curl` failed. If you want failed attempts to be counted (so failedJobsHistoryLimit works and Jobs are marked Failed), do not append `|| echo`. Instead use a form that preserves a non-zero exit code on failure, for example `curl -fS http://todoapp:80/api/health` (or `curl -fsS ...`) so the container exits with non-zero status when the endpoint is unreachable. This aligns with the intent of tracking successful/failed job history.